core/src/main/scala/chisel3/Connectable.scala

```diff
@@ -0,0 +1,486 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3
+
+import chisel3.internal.{prefix, BiConnect, Builder}
+import chisel3.internal.Builder.pushCommand
+import chisel3.internal.firrtl._
+import chisel3.internal.sourceinfo.SourceInfo
+import chisel3.experimental.{Analog, DataMirror}
+import chisel3.experimental.Defaulting._
+
+import scala.collection.mutable
+import chisel3.internal.ChildBinding
+
+/** The default connection operators for Chisel hardware components */
+object Connectable {
+
+  /** ConnectableData Typeclass defines the following operators on all subclasses of Data: :<=, :>=, :<>=, :#=
+    *
+    * @param consumer the left-hand-side of the connection
+    */
+  implicit class ConnectableData[T <: Data](consumer: T) {
+
+    /** The "aligned connection operator" between a producer and consumer.
+      *
+      * For `consumer :<= producer`, each of consumer's leaf fields WHO ARE ALIGNED WITH RESPECT TO CONSUMER are driven from the corresponding producer leaf field
+      * All producer's leaf/branch alignments (with respect to producer) do not influence the connection.
+      *
+      * The following restrictions apply:
+      *  - The Chisel type of consumer and producer must be the "same shape" recursively:
+      *    - All ground types are the same (UInt and UInt are same, SInt and UInt are not), but widths can be different
+      *    - All vector types are the same length
+      *    - All bundle types have the same field names, but the flips of fields can be different between producer and consumer
+      *  - The leaf fields that are ultimately assigned to, must be assignable. This means they cannot be module inputs or instance outputs.
+      *
+      * @note Connecting two [[Decoupled]]'s would connect `bits` and `valid` from producer to consumer, but leave `ready` unconnected
+      * @note If the widths differ between consumer/producer, the assignment will still occur and truncation, if necessary, is implicit
+      *
```

Comment:
by the way, at least in my `sbt doc` converter to HTML, multiple @notes just get rendered as a big block of text with commas between them 🤷